tests/features/common.py

```diff
@@ -0,0 +1,156 @@
+        assert _is_static_method(SomeClass, "static_method") is True
+        assert _is_static_method(SomeClass, "instance_method") is False
+
+        some_class = SomeClass()
+        assert _is_static_method(some_class, "class_method") is False
+        assert _is_static_method(some_class, "static_method") is True
+        assert _is_static_method(some_class, "instance_method") is False
+
+        assert _is_static_method(DerivedClass, "class_method") is False
+        assert _is_static_method(DerivedClass, "static_method") is True
+        assert _is_static_method(DerivedClass, "instance_method") is False
+
+        derived_class = DerivedClass()
+        assert _is_static_method(derived_class, "class_method") is False
+        assert _is_static_method(derived_class, "static_method") is True
+        assert _is_static_method(derived_class, "instance_method") is False
+
+    def test_is_static_method_proxied(self):
+        # pylint: disable=not-callable,invalid-name
+        SomeClassProxy = Proxy(SomeClass)
+        assert _is_static_method(SomeClassProxy, "class_method") is False
+        assert _is_static_method(SomeClassProxy, "static_method") is True
+        assert _is_static_method(SomeClassProxy, "instance_method") is False
+
+        some_class = SomeClassProxy()
+        assert _is_static_method(some_class, "class_method") is False
+        assert _is_static_method(some_class, "static_method") is True
+        assert _is_static_method(some_class, "instance_method") is False
+
+        DerivedClassProxy = Proxy(DerivedClass)
+        assert _is_static_method(DerivedClassProxy, "class_method") is False
+        assert _is_static_method(DerivedClassProxy, "static_method") is True
+        assert _is_static_method(DerivedClassProxy, "instance_method") is False
+
+        derived_class = DerivedClassProxy()
+        assert _is_static_method(derived_class, "class_method") is False
+        assert _is_static_method(derived_class, "static_method") is True
+        assert _is_static_method(derived_class, "instance_method") is False
+
+    def test_isproperty(self):
+        class BaseClass:
+            @property
+            def method1(self):
+                pass
+
+            def method2(self):
+                pass
+
+        class ChildClass(BaseClass):
+            pass
+
+        base_instance = BaseClass()
+        child_instance = ChildClass()
+        assert _isproperty(base_instance, "method1") is True
+        assert _isproperty(base_instance, "method2") is False
+        assert _isproperty(BaseClass, "method1") is True
+        assert _isproperty(BaseClass, "method2") is False
+        assert _isproperty(child_instance, "method1") is True
+        assert _isproperty(child_instance, "method2") is False
+        assert _isproperty(ChildClass, "method1") is True
+        assert _isproperty(ChildClass, "method2") is False
+        assert _isproperty(Mock(), "method2") is False
```

Comment:
I think those could be made into smaller classes also, that would be parents of this one. Not necessarily, but I feel like those tests are more like a series of related tests. I just don't know where we should draw the line :smile: 
I've actually been thinking this for a couple of other tests with similar initialization (creation of similar classes) but didn't write anything because I am not sure if the tests would leak (though it shouldn't if the teardown tests pass)

Reply:
Yeah, I guess we could use common classes for everything, but might be good to have a few tests also that don't use the common classes if teardown doesn't work properly.